rust/kernel/sync/mutex.rs

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: GPL-2.0
+
+use super::{Guard, Lock};
+use crate::{bindings, CStr};
+use core::{cell::UnsafeCell, pin::Pin};
+
+/// Safely initialises a [`Mutex`] wih the given name and new lock class.
+#[macro_export]
+macro_rules! mutex_init {
+    ($mutex:expr, $name:literal) => {
+        $crate::init_with_lockdep!($mutex, $name)
+    };
+}
+
+/// Exposes the kernel's `struct mutex`. When multiple threads attempt to lock the same mutex, only
```

Comment:
```suggestion
/// Exposes the kernel's [`struct mutex`]. When multiple threads attempt to lock the same mutex, only
```

plus at the bottom:

```
/// [`struct mutex`]: ../../../../include/linux/mutex.h
```

I might have got the number of `..`s wrong :-)

Reply:
Done.